src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.cs

```diff
@@ -2645,6 +2645,68 @@ public static void Sort<T>(this Span<T> span, Comparison<T> comparison)
             }
         }
 
+        /// <summary>
+        /// Replaces all occurrences of <paramref name="oldValue"/> with <paramref name="newValue"/>.
+        /// </summary>
+        /// <typeparam name="T">The type of the elements in the span.</typeparam>
+        /// <param name="span">The span in which the elements should be replaced.</param>
+        /// <param name="oldValue">The value to be replaced with <paramref name="newValue"/>.</param>
+        /// <param name="newValue">The value to replace all occurrences of <paramref name="oldValue"/>.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void Replace<T>(this Span<T> span, T oldValue, T newValue) where T : IEquatable<T>?
+        {
+            if (SpanHelpers.CanVectorizeAndBenefit<T>(span.Length))
+            {
+                nuint length = (uint)span.Length;
+
+                if (Unsafe.SizeOf<T>() == sizeof(byte))
+                {
+                    ref byte src = ref Unsafe.As<T, byte>(ref MemoryMarshal.GetReference(span));
+                    SpanHelpers.ReplaceValueType(
+                        ref src,
+                        ref src,
+                        Unsafe.As<T, byte>(ref oldValue),
+                        Unsafe.As<T, byte>(ref newValue),
+                        length);
+                }
+                else if (Unsafe.SizeOf<T>() == sizeof(ushort))
+                {
+                    // Use ushort rather than short, as this avoid a sign-extending move.
```

Comment:
Should we be using the unsigned types in general in these SpanHelper delegations across all of MemoryExtensions, unless there's a good reason to do otherwise?

Reply:
I don't remember exactly why, but anywhere in the JIT-dasm I spotted the `movsxd` and that got eliminated by using `ushort`.

Besides this one only the IndexOfAnyInRange-variants use `ushort`.
One would need to review all other places, not that regression may occur. Shall I do that? (If so, I can look into this not earlier than next week).

Reply:
FYI [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAJI8ovLrl5hcAbho1iAZiakGAYQYBvGg11NlANRhgM0cqQAcAHlwALaBgB8DALIAKOw4YA3AJQMAXmcjEzNLFjVYbAwYNz85aj19BhDTKHNrDk8oJ1c3LPscn38glOM0jIiomLjfBIBfIA=).
Note the difference
```diff
- L0003: movsx rax, r8w
+ L0003: movzx eax, r8w
```
This is simple repo, but I remember to have seen this in the actual code too (and I don't like the `movsx`s if not needed.